config/filter.d/assp.conf

```diff
@@ -1,24 +1,24 @@
-# Fail2Ban filter for Anti-Spam SMTP Proxy Server also known as ASSP
+# Fail2Ban filter for Anti-Spam SMTP Proxy Server (ASSP) Version 2.5.1 (or later)
 # 
-#    Honmepage:   http://www.magicvillage.de/~Fritz_Borgstedt/assp/0003D91C-8000001C/
-#    ProjektSite: http://sourceforge.net/projects/assp/?source=directory
+#    Homepage:    http://sourceforge.net/projects/assp/
+#    ProjectSite: http://sourceforge.net/projects/assp/?source=directory
 #
 #
 
 [Definition] 
 
-__assp_actions = (?:dropping|refusing)
-
-failregex = ^(:? \[SSL-out\])? <HOST> max sender authentication errors \(\d{,3}\) exceeded -- %(__assp_actions)s connection - after reply: \d{3} \d{1}\.\d{1}.\d{1} Error: authentication failed: \w+;$
-			^(?: \[SSL-out\])? <HOST> SSL negotiation with client failed: SSL accept attempt failed with unknown error.*:unknown protocol;$
-			^ Blocking <HOST> - too much AUTH errors \(\d{,3}\);$
+failregex = <HOST> \<\S+@\S+\.\S+\> to: \S+@\S+\.\S+ relay attempt blocked for: \S+$
+            <HOST> \[SMTP Error\] 535 5\.7\.8 Error: authentication failed.*$
```

Comment:
It isn't my call but I will give my two cents: I would say no. ASSP is a spam firewall and it is useless if not kept current. As a security tool fail2ban should only support safe supported versions and clearly list that as a requirement. That is why I explicitedly listed version 2.5.1 or later in the config file. Any supported version of ASSP will not generate those old patterns. I also suspect the old regex actually was wrong in that it would fail to match lines where the flags were different. My regex is deliberately not anchored. The log lines have a whole list of flags which may or may not appear such as [SSL-in] [TLS-in] [TLS-out] [RelayAttempt] etc. I also believe the order can change. There were also other patterns which varied such as m1-52131-09651 at the start of the line. My attempts at anchoring the patterns caused many missed attacks during my testing. The reason for the .* is other content sometimes follows. I could just remove .*$ if you want. Some examples:
Feb-10-16 15:06:02 [Worker_1] [TLS-out] 0.0.0.0 [SMTP Error] 535 5.7.8 Error: authentication failed: UGFzc3dvcmQ6
Feb-10-16 15:06:12 [Worker_1] [TLS-out] 0.0.0.0 [SMTP Error] 535 5.7.8 Error: authentication failed: Connection lost to authentication server
Feb-10-16 15:06:12 [Worker_1] [TLS-out] 0.0.0.0 [SMTP Error] 535 5.7.8 Error: authentication failed: